Date: 2005-10-12
Initial Package Version: 2.9.7
Upstream Status: Not yet submitted
Origin: Miklos Vajna <vmiklos@frugalware.org>
Description: Adds distcc support to makepkg.
 etc/makepkg.conf |    7 +++++++
 scripts/makepkg  |    5 +++++
 2 files changed, 12 insertions(+)
diff -Naur pacman-2.9.7.orig/etc/makepkg.conf pacman-2.9.7/etc/makepkg.conf
--- pacman-2.9.7.orig/etc/makepkg.conf	2005-10-12 00:09:24.000000000 +0200
+++ pacman-2.9.7/etc/makepkg.conf	2005-10-12 00:14:22.000000000 +0200
@@ -33,6 +33,13 @@
 # Core package list to be installed in the chroot.
 export COREPKGS="core chroot-core devel-core"
 
+# Enable distributed C/C++/ObjC compiler
+# Don't forget to modify MAKEFLAGS to include -jN, where N is twice the
+# number of CPUs +1 available
+export DISTCC="n"
+# Space-separated list of volunteer host specifications.
+export DISTCC_HOSTS=""
+
 # Enable colorized output messages
 export USE_COLOR="y"
 
diff -Naur pacman-2.9.7.orig/scripts/makepkg pacman-2.9.7/scripts/makepkg
--- pacman-2.9.7.orig/scripts/makepkg	2005-10-12 00:09:24.000000000 +0200
+++ pacman-2.9.7/scripts/makepkg	2005-10-12 00:15:17.000000000 +0200
@@ -944,6 +944,15 @@
 	exit 0
 fi
 
+# use distcc if requested
+if [ "$DISTCC" = "y" ]; then
+	[ -d /usr/lib/distcc/bin ] && export PATH=/usr/lib/distcc/bin:$PATH
+	if [ "$INCHROOT" = "1" ]; then
+		[ -d /var/tmp/fst/.distcc ] || mkdir /var/tmp/fst/.distcc
+		export DISTCC_DIR=/var/tmp/fst/.distcc
+	fi
+fi
+
 # use ccache if it's available
 if [ "$NOCCACHE" = "0" ]; then
 	[ -d /usr/lib/ccache/bin ] && export PATH=/usr/lib/ccache/bin:$PATH
